src/neo.c

```diff
@@ -410,7 +410,16 @@ static void to_address(char *dest, unsigned int dest_len, const unsigned char *s
     memmove(address + 1 + SCRIPT_HASH_LEN, address_hash_result_1, SCRIPT_HASH_CHECKSUM_LEN);
 
     // encode the version + address + cehcksum in base58
-    encode_base_58(address, ADDRESS_LEN, dest, dest_len);
+    unsigned int encode_len = encode_base_58(address, ADDRESS_LEN, dest, dest_len);
+
+    // Add a null terminator to the end of the string.
+    // Base 58 address length should be inferior to dest_len so we can safely add the null terminator.
+    if (encode_len < dest_len) {
+        dest[encode_len] = '\0';
```

Comment:
This is probably unnecessary as base 58 NEO addresses are 34 bytes long and the dest_len will always be a 35 bytes `0` initialized array (per line 897) : `char address_base58[ADDRESS_BASE58_LEN + 1] = {0};`

But we never know I guess ?

Reply:
I should be careful to make sure we only input [ADDRESS_BASE58_LEN + 1] length arrays to the `to_address` function now !